sdk/samples/iot/paho_iot_hub_methods_sample.c

```diff
@@ -325,7 +326,7 @@ static void send_method_response(
           rc = az_iot_hub_client_methods_response_get_publish_topic(
               &hub_client,
               method_request->request_id,
-              status,
+              (uint16_t)status,
```

Comment:
Line 354 below is missing a similar cast:
```C
IOT_SAMPLE_LOG("Status: %u", status);
```

That's causing the CI failure:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=515293&view=logs&j=d0a3aede-fed8-508e-bfcc-69b7c0c9b6ed&t=68448d55-6b50-5f76-d0ee-4facc07a4ed2&l=118
```text
paho_iot_hub_methods_sample.c
D:\a\1\s\sdk\samples\iot\paho_iot_hub_methods_sample.c(356,1): error C2220: the following warning is treated as an error [D:\a\1\s\build\sdk\samples\iot\paho_iot_hub_methods_sample.vcxproj]
D:\a\1\s\sdk\samples\iot\paho_iot_hub_methods_sample.c(354): warning C6340: Mismatch on sign: 'enum az_iot_status' passed as _Param_(2) when some unsigned type is required in call to 'printf'. [D:\a\1\s\build\sdk\samples\iot\paho_iot_hub_methods_sample.vcxproj]
```

Reply:
thank you